openedx/core/djangoapps/programs/models.py

```diff
@@ -42,3 +52,13 @@ def is_student_dashboard_enabled(self):
         be enabled or not.
         """
         return self.enabled and self.enable_student_dashboard
+
+    @property
+    def is_cache_enabled(self):
+        """Indicate whether programs api will use the cache or not."""
+        return self.enabled and self.cache_ttl > 0
+
+    @property
+    def programs_cache_settings(self):
```

Comment:
@awais786 I would consider removing this method.  Since the cache key is hardcoded, the utils module could simply import it directly from this module, and access the `cache_ttl` value of the config model directly.  This would simplify the new functions in utils and reduce the amount of test code needed.